Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadColorBaker.cs

```diff
@@ -0,0 +1,78 @@
+using Autodesk.AutoCAD.Colors;
+using Microsoft.Extensions.Logging;
+using Speckle.Sdk;
+using Speckle.Sdk.Models.Proxies;
+using AutocadColor = Autodesk.AutoCAD.Colors.Color;
+
+namespace Speckle.Connectors.Autocad.HostApp;
+
+/// <summary>
+/// Expects to be a scoped dependency for a given operation and helps with layer creation and cleanup.
+/// </summary>
+public class AutocadColorBaker
+{
+  private readonly ILogger<AutocadColorBaker> _logger;
+
+  public AutocadColorBaker(ILogger<AutocadColorBaker> logger)
+  {
+    _logger = logger;
+  }
+
+  /// <summary>
+  /// For receive operations
+  /// </summary>
+  public Dictionary<string, AutocadColor> ObjectColorsIdMap { get; } = new();
+
+  /// <summary>
+  /// Parse Color Proxies and stores in ObjectColorIdMap the relationship between object ids and colors
+  /// </summary>
+  /// <param name="colorProxies"></param>
+  /// <param name="onOperationProgressed"></param>
+  public void ParseColors(List<ColorProxy> colorProxies, Action<string, double?>? onOperationProgressed)
+  {
+    var count = 0;
+    foreach (ColorProxy colorProxy in colorProxies)
+    {
+      try
+      {
+        onOperationProgressed?.Invoke("Converting colors", (double)++count / colorProxies.Count);
+
+        // skip any colors with source = layer, since object color default source is by layer
+        if (colorProxy["source"] is string source && source == "layer")
+        {
+          continue;
+        }
+
+        foreach (string objectId in colorProxy.objects)
+        {
+          AutocadColor convertedColor = ConvertColorProxyToColor(colorProxy);
+#if NET8_0
+          ObjectColorsIdMap.TryAdd(objectId, convertedColor);
+#else
+          if (!ObjectColorsIdMap.ContainsKey(objectId))
+          {
+            ObjectColorsIdMap.Add(objectId, convertedColor);
+          }
+#endif
+        }
+      }
+      catch (Exception e) when (!e.IsFatal())
+      {
+        _logger.LogError(e, "Failed parsing color proxy."); // TODO: Check with Jedd!
```

Comment:
Only thing here is, Logger templates should not have a trailing `.`

There's a resharper warning about this, unfortunately, there is no rosylin analyser

Also, we have been using `ex` instead of `e` for the variable name of the exception. I think its nice to keep consistent with elsewhere. (tho, I'm not precious about this 😋)